lib/rubygems/spec_fetcher.rb

```diff
@@ -174,20 +174,35 @@ def suggest_gems_from_name(gem_name, type = :latest, num_results = 5)
     max             = gem_name.size / 2
     names           = available_specs(type).first.values.flatten(1)
 
+    min_length = gem_name.length - max
+    max_length = gem_name.length + max
+
     matches = names.map do |n|
```

Comment:
```suggestion
    matches = names.filter_map do |n|
```

Also remove the `.compact` at the end of the loop.

Reply:
Done!